---
format:
  revealjs:
    slide-level: 2
    bibliography:
      - references.yaml
      - references_other.yaml
    csl: ecology.csl
    reference-location: document
    css: styles.css
    slide-number: true
    self-contained: true
---

```{r}
#| label: setup
#| include: false
knitr::opts_chunk$set(echo = TRUE, cache = TRUE)
library(ftolr)
library(ggtree)
library(ggtext)
library(ggdark)
library(scico)
library(patchwork)
library(tidyverse)
library(ape)
library(fontawesome)
```

# Introduction {visibility="hidden"}

## An Open and Continuously <br>Updated Fern Tree of Life<br>(FTOL) {background-image="images/background.png" background-size="contain" background-color="#191919"}

:::: {.columns}

::: {.column width="50%"}

:::: {.medium}
**Joel Nitta<sup>1</sup>**, Eric Schuettpelz<sup>2</sup>, Santiago Ram&iacute;rez-Barahona<sup>3</sup>,  
Wataru Iwasaki<sup>1</sup>
::::

:::: {.small}
1: The University of Tokyo, 2: Smithsonian Institution, 3: Universidad Nacional Aut&oacute;noma de M&eacute;xico  
Botany 2022  
[https://joelnitta.github.io/botany_2022_ftol](https://joelnitta.github.io/botany_2022_ftol)
::::

:::

::: {.column width="50%"}
:::

::::

## Phylogenies are essential to biology

:::: {.columns}

::: {.column width="50%"}
> **Only with a phylogeny** can we begin to understand diversification, regularities in patterns of evolution, or simply suggest individual evolutionary changes within a clade

\- APG

::: {.small}
[http://www.mobot.org/MOBOT/research/APweb/](http://www.mobot.org/MOBOT/research/APweb/)
:::

::: {.tiny}
https://www.digitalatlasofancientlife.org/learn/embryophytes/angiosperms/angiosperm-phylogeny/
:::

:::

::: {.column width="50%"}
![](images/apg.png){height="600"}
:::

::::

## Automated pipelines enable building large trees

::::: {.columns}

:::: {.column width="65%"}
- pyPHLAWD [@Smith2019]
- superSMART [@Antonelli2016]
- superCRUNCH [@Portik2020]
::::

:::: {.column width="10%"}
::::

:::: {.column width="25%" .small}
![](images/supersmart.png){height="400"}
@Antonelli2016
::::

:::::

## Problem 1: tradeoff between scalability and accuracy

- Any automated pipeline must make **shortcuts** and **assumptions**

- **Manual inspection** of all sequences would lead to high-quality results, **but does not scale**

::: {.r-stack}
![](images/balance.png){height="300"}
:::

## Problem 2: tree built today will be out-of-date tomorrow

- Due to the rapid accumulation of data on GenBank

::: {.r-stack}
![](images/time.png){height="300"}
:::

## Our approach: combine automation with customization

Goal: construct a pipeline to generate a **maximally sampled**, **high taxonomic quality** phylogeny of **ferns**

::: {.r-stack}
![](images/conveyer.png){height="300"}
:::

## Why ferns?

A large, diverse, ecologically important group of plants

Much more tractable than seed plants (angiosperms):

- Ferns: ca. 12,000 species, 40-50% sequenced

- Seed plants: ca. 350,000 species, 20% sequenced

::: {.r-stack}
![](images/ferns.png){height="300"}
:::

---

### Sanger-sequenced plastid genes = workhorse of fern molecular systematics

::: {.r-stack}
![](images/genbank_ferns.png){height="550"}
:::

# Methods

---

## GenBank mining

### Sanger: 7 commonly used loci
  - Genes: *atpA*, *atpB*, *matK*, *rbcL*, *rps4*
  - Spacers: *trnL*--*trnF*, *rps4*--*trnS*
  - ca. 5,100 species

### Plastomes
  - 77 single-copy genes + 2 spacers
  - ca. 500 species 

## GenBank mining

- Download data to local database using [restez](https://github.com/ropensci/restez)\* R package

- Use superCRUNCH [@Portik2020] to extract sequences without relying on annotations

![](images/supercrunch.png){height="250"}

::: aside
\*I am current maintainer; if you have questions, please ask!
:::

## Taxonomic name resolution

- Use [World Ferns](https://www.worldplants.de/world-ferns/ferns-and-lycophytes-list) [@Hassler2022] as basis for new, fern-specific taxonomic database, [pteridocat](https://github.com/fernphy/pteridocat)

- Resolve GenBank species names to pteridocat using [taxastand](https://github.com/joelnitta/taxastand)\* R package

::: {.medium}
```{r}
#| label: name-res-ex
#| echo: false

# Read in name resolution results
name_res <- readRDS("data/match_results_resolved_all.RDS")

# Get number of rows
nrow_res <-
name_res %>% 
  nrow() %>%
  scales::number(big.mark = ",")

# Isolate some example rows
name_res %>%
  filter(matched_status == "synonym") %>%
  select(query, matched_name, resolved_name) %>%
  mutate(
    across(everything(), ~nchar(.), .names = "{.col}_nchar"),
    total_nchar = rowSums(across(contains("_nchar"))),
    q_genus = str_split(query, " ") %>% map_chr(1)
    ) %>%
  filter(total_nchar < 70) %>%
  group_by(q_genus) %>%
  slice_min(order_by = total_nchar) %>%
  ungroup() %>%
  arrange(total_nchar) %>%
  slice(1:2) %>%
  arrange(query) %>%
  select(query, matched_name, resolved_name) %>%
  knitr::kable()
```

... (`r nrow_res` total)
:::

::: aside
\*Check out talk in Biodiversity Informatics & Herbarium Digitization
:::

## Automated cleaning

- Run all-by-all BLAST [@Camacho2009]

- Any query matching the wrong family is excluded as mis-ID

::: {.small}
|species|accession|locus|query family| match family |
| ----- | ------- | --- | ------- | ----- |
Abacopteris_gymnopteridifrons | JF303974 | rbcL | Thelypteridaceae | Athyriaceae
Angiopteris_evecta | AY344778 | trnL-trnF | Marattiaceae | Ophioglossaceae
...
(70 total)
:::

## Phylogenetic analysis: backbone

- Align plastome sequences with MAFFT [@Katoh2002] (544 species x 74,883 bp, 12.1% missing)

- Infer tree using ML in IQ-TREE [@Nguyen2015] (concatenated matrix, no paritioning)

## Phylogenetic analysis: full tree

- Align Sanger sequences with MAFFT (5,582 species x 12,716 bp, 77% missing)

- Infer tree in IQ-TREE (concatenated matrix, no paritioning) **with plastome tree as constraint**

::: {.fragment}
- Before final analysis, run IQ-TREE in "fast" mode and manually inspect for rogues
:::

::: {.fragment}
- Date tree with treePL [@Smith2012]
:::

---

![](images/workflow.png){.r-stretch}

::: aside
  - Control workflow with **targets** R package [@Landau2021]
  - Runtime ca. 10--14 days
:::

# Results

## {background-image="images/cov_tree.png" background-size="contain"}

```{r}
#| label: make-bs-plot
#| echo: false
#| run: true
#| message: false
#| warning: false

phy <- ftolr::ft_tree(drop_og = TRUE)

# Make a dataframe (tibble) with node IDs (integers) and their corresponding
# bootstrap support values. The tibble has two columns: one called "node",
# the other can be named as we like (here, "bootstrap")
bs_tibble <- tibble(
  # hard-code node ID: internal nodes start after tip nodes,
  # and phy$node.label is in the same order as internal nodes
  node = seq_len(Nnode(phy)) + Ntip(phy),
  bootstrap = parse_number(phy$node.label))

# Use the ggtree::`%<+%` operator to map the bootstrap values onto the tree

bs_tree_plot <- ggtree(
    phy, layout = "fan", size = 0.25, color = "grey70") %<+% bs_tibble +
  geom_nodepoint(aes(color = bootstrap), size = 0.75) +
  scale_color_viridis_c(option = "C") +
  labs(color = "BS") +
  theme(
    panel.background = element_rect(fill = "grey10", colour = NA),
    plot.background = element_rect(fill = "grey10", colour = NA),
    legend.position = "none")

# export image to working dir, then edit in powerpoint. too hard to get
# sizes to work as desired in Rmarkdown
ggsave(
  plot = bs_tree_plot, file = "working/bs_tree_plot.png",
  width = 10, height = 10, units = "in")

# Also make distribution plot of node support values
support <- tibble(
  bs = parse_number(phy$node.label)) %>%
  filter(!is.na(bs))

support_counts <-
support %>%
  mutate(bin = cut_width(bs, width = 1, center = 0.5)) %>%
  group_by(bin) %>%
  summarize(
    count = n(),
    .groups = "drop"
  ) %>%
  mutate(
    bin_start = str_match(bin, "(\\d+)\\,") %>%
      magrittr::extract(,2) %>%
      parse_number()) %>%
  mutate(bin_start = bin_start / 100)

n_nodes <- nrow(support) %>% scales::number(big.mark = ",")

# Dark version
bs_histogram <- ggplot(support_counts, aes(x = bin_start, y = count, fill = bin_start)) +
  geom_col() +
  scale_x_continuous(labels = scales::percent) +
  scale_fill_viridis_c(option = "C", labels = scales::percent) +
  labs(
    x = "BS",
    y = "Count",
    fill = "BS"
  ) +
  dark_theme_gray(base_size = 24) +
  theme(
    plot.subtitle = element_markdown(),
    panel.background = element_rect(fill = "grey10", colour = NA),
    plot.background = element_rect(fill = "grey10", colour = NA),
    legend.background = element_rect(fill = "grey10", colour = NA),
    legend.position = "right")

ggsave(
  plot = bs_histogram, file = "working/bs_histogram.png")
```

## {background-image="images/bs_tree.png" background-size="contain"}

## Revisiting the timeline of fern diversification

- 51 fossils (**2x more than previous**)

- Pushes back stem ages for most families ca. 10-30 my

- Suggests ferns **did not diversify "in the shadow" of angiosperms**

---
## Older stem<br>ages for most<br>families

![](images/stem_ages.png){.absolute top=0 left=400 height="700"}

## Web portal

[https://fernphy.github.io/](https://fernphy.github.io/){preview-link="true"}

- Data downloads

- Shiny app for exploring data

## R package **ftolr**

[https://github.com/fernphy/ftolr](https://github.com/fernphy/ftolr){preview-link="true"}

- Read tree and data (alignments) directly into R

- Options for outgroups, rooting, locus selection, etc.

```{r}
#| label: ftolr-demo
#| echo: true

library(ftolr)
ft_tree(drop_og = TRUE)
```

## Community involvement

Consulted with a taxonomic expert on family Thelypteridaceae (S. Fawcett) between v1.0.0 and v1.1.0

- Number of non-monophyletic genera **dropped from 16 to 7**

# Summary

## FTOL hits sweet spot between automation and customization

- Automated, versioned mining of GenBank data

- Custom taxonomy tailored for ferns

- Input from taxonomic experts and broader community

- **Model for other plant groups at similar scale?**

## Future directions

- Completion of FTOL
   
   - "Unlock the vault" of **herbarium specimens** via plastome skimming

- Integration with **Pteridophyte Phylogeny Group II**
    
- Transition to **phylogenomics** for all species

## Acknowledgements

- Japan Society for the Promotion of Science

- Smithsonian National Museum of Natural History Peter Buck Fellowship

- Members of the Iwasaki lab, The University of Tokyo

- A.E. White

- S. Fawcett

- M. Hassler

## hidden refs {visibility="hidden"}

[@Testo2016a]

## References {.scrollable}